lib/web/controllers/base.go

```diff
@@ -1,20 +1,38 @@
-package controllers
+package webcontrollers
 
 import (
 	"runtime"
 	"strings"
 
 	"go-google-scraper-challenge/helpers"
+	web_helpers "go-google-scraper-challenge/helpers/web"
 	api_controllers "go-google-scraper-challenge/lib/api/v1/controllers"
 
 	"github.com/gin-gonic/gin"
 )
 
-type BaseWebController struct {
+type BaseController struct {
 	api_controllers.BaseController
 }
 
-func (c *BaseWebController) Data(ctx *gin.Context, data gin.H) gin.H {
+func (c *BaseController) EnsureGuestUser(ctx *gin.Context) {
```

Comment:
Should we convert these 2 functions into middleware and add them to a router group instead?
You can find the example in [varun-api](https://github.com/nimblehq/varun-api/blob/develop/lib/api/v1/routers/router.go) or [recruitmate-web](https://github.com/nimblehq/recruitmate-web/blob/development/lib/web/routers/router.go)

Reply:
@hoangmirs Actually I planned to work on it in #96 maybe I should work on it soon 😅